sdks/java/core/src/main/java/org/apache/beam/sdk/io/TextIO.java

```diff
@@ -448,7 +448,15 @@ private Read() {}
       return new Bound<>(DEFAULT_TEXT_CODER).withoutValidation();
     }
 
-    // TODO: appendingNewlines, header, footer, etc.
+    public static Bound<String> withHeader(String header) {
```

Comment:
Add javadoc in similar style to withCoder/withoutValidation
State that setting header/footer to null clears the set header/footer